In [1]:
import pandas as pd
import matplotlib.pyplot as plt



In dit bestand zal worden gekeken in hoever "Vrouwen en kinderen eerst" gold bij deze selectie van de passagies van de Titanic, en dus hoeveel van de overlevenden vrouwen en kinderen (<18) zijn. Hieronder volgt eerst de hele dataset.

In [2]:
titanic = pd.read_csv("titanic.csv")
print(titanic)

     Survived  Pclass                                               Name  \
0           0       3                             Mr. Owen Harris Braund   
1           1       1  Mrs. John Bradley (Florence Briggs Thayer) Cum...   
2           1       3                              Miss. Laina Heikkinen   
3           1       1        Mrs. Jacques Heath (Lily May Peel) Futrelle   
4           0       3                            Mr. William Henry Allen   
..        ...     ...                                                ...   
882         0       2                               Rev. Juozas Montvila   
883         1       1                        Miss. Margaret Edith Graham   
884         0       3                     Miss. Catherine Helen Johnston   
885         1       1                               Mr. Karl Howell Behr   
886         0       3                                 Mr. Patrick Dooley   

        Sex   Age  Siblings/Spouses Aboard  Parents/Children Aboard     Fare  
0      m

In [3]:
print(titanic.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 887 entries, 0 to 886
Data columns (total 8 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Survived                 887 non-null    int64  
 1   Pclass                   887 non-null    int64  
 2   Name                     887 non-null    object 
 3   Sex                      887 non-null    object 
 4   Age                      887 non-null    float64
 5   Siblings/Spouses Aboard  887 non-null    int64  
 6   Parents/Children Aboard  887 non-null    int64  
 7   Fare                     887 non-null    float64
dtypes: float64(2), int64(4), object(2)
memory usage: 55.6+ KB
None




Dit zijn de overlevenden:

In [4]:
overleefd = titanic[titanic["Survived"] == 1]
print(overleefd)

     Survived  Pclass                                               Name  \
1           1       1  Mrs. John Bradley (Florence Briggs Thayer) Cum...   
2           1       3                              Miss. Laina Heikkinen   
3           1       1        Mrs. Jacques Heath (Lily May Peel) Futrelle   
8           1       3   Mrs. Oscar W (Elisabeth Vilhelmina Berg) Johnson   
9           1       2                 Mrs. Nicholas (Adele Achem) Nasser   
..        ...     ...                                                ...   
871         1       3                           Miss. Adele Kiamie Najib   
875         1       1       Mrs. Thomas Jr (Lily Alexenia Wilson) Potter   
876         1       2        Mrs. William (Imanita Parrish Hall) Shelley   
883         1       1                        Miss. Margaret Edith Graham   
885         1       1                               Mr. Karl Howell Behr   

        Sex   Age  Siblings/Spouses Aboard  Parents/Children Aboard     Fare  
1    fem



Dit zijn de overledenen:

In [5]:
overleden = titanic[titanic["Survived"] == 0]
print(overleden)

     Survived  Pclass                                 Name     Sex   Age  \
0           0       3               Mr. Owen Harris Braund    male  22.0   
4           0       3              Mr. William Henry Allen    male  35.0   
5           0       3                      Mr. James Moran    male  27.0   
6           0       1               Mr. Timothy J McCarthy    male  54.0   
7           0       3        Master. Gosta Leonard Palsson    male   2.0   
..        ...     ...                                  ...     ...   ...   
880         0       3                Mr. Henry Jr Sutehall    male  25.0   
881         0       3  Mrs. William (Margaret Norton) Rice  female  39.0   
882         0       2                 Rev. Juozas Montvila    male  27.0   
884         0       3       Miss. Catherine Helen Johnston  female   7.0   
886         0       3                   Mr. Patrick Dooley    male  32.0   

     Siblings/Spouses Aboard  Parents/Children Aboard     Fare  
0                     

Hieronder wordt de tabel gesplitst in overleden en overlevende mannen, vrouwen en kinderen.

In [6]:
overleefdmannen = overleefd[(overleefd["Sex"] == "male") & (overleefd["Age"] >= 18)]
overleefdvrouwen = overleefd[(overleefd["Sex"] == "female") & (overleefd["Age"] >= 18)]
overleefdkinderen = overleefd[overleefd["Age"] <= 18]
overledenmannen = overleden[(overleden["Sex"] == "male") & (overleden["Age"] >= 18)]
overledenvrouwen = overleden[(overleden["Sex"] == "female") & (overleden["Age"] >= 18)]
overledenkinderen = overleden[overleden["Age"] <= 18]

In [7]:
print("Aantal overlevende mannen: " + str(len(overleefdmannen)))
print("Aantal overlevende vrouwen: " + str(len(overleefdvrouwen)))
print("Aantal overlevende kinderen: " + str(len(overleefdkinderen)))
print("")
print("Aantal overleden mannen: " + str(len(overledenmannen)))
print("Aantal overleden vrouwen: " + str(len(overledenvrouwen)))
print("Aantal overleden kinderen: " + str(len(overledenkinderen)))
print("")
print("Totaal aantal mannen: " + str(len(overleefdmannen) + len(overledenmannen)))
print("Totaal aantal vrouwen: " + str(len(overleefdvrouwen) + len(overledenvrouwen)))
print("Totaal aantal kinderen: " + str(len(overleefdkinderen) + len(overledenkinderen)))
print("")
print("Aantal overlevenden: " + str(len(overleefd)))
print("Aantal overledenen: " + str(len(overleden)))
print("")
print("Percentages binnen de overlevenden:")
print("Overlevende mannen: " + str(round(len(overleefdmannen) / len(overleefd) * 100, 2)) + " %")
print("Overlevende vrouwen: " + str(round(len(overleefdvrouwen) / len(overleefd) * 100, 2)) + " %")
print("Overlevende kinderen: " + str(round(len(overleefdkinderen) / len(overleefd) * 100, 2)) + " %")
print("")
print("Percentages binnen de overledenen:")
print("Overleden mannen: " + str(round(len(overledenmannen) / len(overleden) * 100, 2)) + " %")
print("Overleden vrouwen: " + str(round(len(overledenvrouwen) / len(overleden) * 100, 2)) + " %")
print("Overleden kinderen: " + str(round(len(overledenkinderen) / len(overleden) * 100, 2)) + " %")
print("")
print("Overlevingskans:")
totaalmannen = len(overleefdmannen) + len(overledenmannen)
totaalvrouwen = len(overleefdvrouwen) + len(overledenvrouwen)
totaalkinderen = len(overleefdkinderen) + len(overledenkinderen)
print("Overlevingskans mannen: " + str(round(len(overleefdmannen) / totaalmannen * 100, 2)) + " %")
print("Overlevingskans vrouwen: " + str(round(len(overleefdvrouwen) / totaalvrouwen * 100, 2)) + " %")
print("Overlevingskans kinderen: " + str(round(len(overleefdkinderen) / totaalkinderen * 100, 2)) + " %")

Aantal overlevende mannen: 84
Aantal overlevende vrouwen: 193
Aantal overlevende kinderen: 78

Aantal overleden mannen: 422
Aantal overleden vrouwen: 58
Aantal overleden kinderen: 88

Totaal aantal mannen: 506
Totaal aantal vrouwen: 251
Totaal aantal kinderen: 166

Aantal overlevenden: 342
Aantal overledenen: 545

Percentages binnen de overlevenden:
Overlevende mannen: 24.56 %
Overlevende vrouwen: 56.43 %
Overlevende kinderen: 22.81 %

Percentages binnen de overledenen:
Overleden mannen: 77.43 %
Overleden vrouwen: 10.64 %
Overleden kinderen: 16.15 %

Overlevingskans:
Overlevingskans mannen: 16.6 %
Overlevingskans vrouwen: 76.89 %
Overlevingskans kinderen: 46.99 %


Het ziet ernaar uit dat binnen deze dataset de overlevingskans voor vrouwen het hoogste is, daarna voor kinderen en ten slotte is deze het laagst voor mannen (waar er wel meer dan twee keer zoveel van zijn dan vrouwen, maar meer dan de helft van de overlevenden is vrouw). Uiteraard gaat deze dataset niet over alle passagiers (2223, de dataset gaat over 887 passagiers), maar het lijkt erop dat het voor deze keer een voordeel is om een vrouw te zijn in 1912 (en natuurlijk hoef je dan over een paar jaar niet te vechten in de Eerste Wereldoorlog, of als je in 1912 kind bent in de Tweede). "Vrouwen en kinderen eerst" is dus ook echt zo in levensbedreigende situaties.